RedacteurPortaal.Grains/Grains/ArchiveGrain.cs

```diff
@@ -17,37 +20,201 @@ public class ArchiveGrain : Grain, IArchiveGrain
         this.archive = archive;
     }
 
-    public async Task Delete()
+    public Task<ArchiveModel> Get()
     {
-        await this.archive.ClearStateAsync();
+        var state = this.archive.State;
+        return Task.FromResult(state);
+    }
+
+    public async Task<MediaVideoItem> GetVideoItem(Guid guid)
+    {
+        var videoItems = this.archive.State.MediaVideoItems;
+        var videoItem = videoItems.Find(x => x.Id.Equals(guid));
```

Comment:
```suggestion
        var videoItem = videoItems.Single(x => x.Id.Equals(guid));
        
        await return Task.FromResult(videoItem)
```

Reply:
Single() throwd automatisch als hij niks kan vinden, en als er meer dan 1 resultaat is